controllers/sysbench/job.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2019 The xridge kubestone contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sysbench
+
+import (
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/firepear/qsplit"
+	perfv1alpha1 "github.com/xridge/kubestone/api/v1alpha1"
+)
+
+// NewJob creates a sysbench benchmark job
+func NewJob(cr *perfv1alpha1.Sysbench) *batchv1.Job {
+	labels := map[string]string{
+		"app":               "sysbench",
+		"kubestone-cr-name": cr.Name,
+	}
+	for key, value := range cr.Spec.PodLabels {
+		labels[key] = value
+	}
+
+	sysbenchCmdLineArgs := []string{}
+	sysbenchCmdLineArgs = append(sysbenchCmdLineArgs, qsplit.ToStrings([]byte(cr.Spec.Options))...)
+	sysbenchCmdLineArgs = append(sysbenchCmdLineArgs, cr.Spec.TestName, cr.Spec.Command)
+
+	backoffLimit := int32(0)
+
+	job := batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cr.Name,
+			Namespace: cr.Namespace,
+		},
+		Spec: batchv1.JobSpec{
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels,
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:            "sysbench",
+							Image:           cr.Spec.Image.Name,
+							ImagePullPolicy: corev1.PullPolicy(cr.Spec.Image.PullPolicy),
+							Command:         []string{"/usr/local/bin/sysbench"},
```

Comment:
I'd suggest not to override the original command:  then the creator of the image creator can decide how the container should be executed.

Reply:
The Dockerfile does not contain an ENTRYPOINT, only CMD is set. https://github.com/xridge/sysbench-docker/blob/master/Dockerfile#L46

```sh
$ kubectl  explain pods.spec.containers.command
KIND:     Pod
VERSION:  v1

FIELD:    command <[]string>

DESCRIPTION:
     Entrypoint array. Not executed within a shell. The docker image's
     ENTRYPOINT is used if this is not provided. Variable references $(VAR_NAME)
     are expanded using the container's environment. If a variable cannot be
     resolved, the reference in the input string will be unchanged. The
     $(VAR_NAME) syntax can be escaped with a double $$, ie: $$(VAR_NAME).
     Escaped references will never be expanded, regardless of whether the
     variable exists or not. Cannot be updated. More info:
     https://kubernetes.io/docs/tasks/inject-data-application/define-command-argument-container/#running-a-command-in-a-shell```

Reply:
The Dockerfile does not contain an ENTRYPOINT, only CMD is set. https://github.com/xridge/sysbench-docker/blob/master/Dockerfile#L46

```sh
$ kubectl  explain pods.spec.containers.command
KIND:     Pod
VERSION:  v1

FIELD:    command <[]string>

DESCRIPTION:
     Entrypoint array. Not executed within a shell. The docker image's
     ENTRYPOINT is used if this is not provided. Variable references $(VAR_NAME)
     are expanded using the container's environment. If a variable cannot be
     resolved, the reference in the input string will be unchanged. The
     $(VAR_NAME) syntax can be escaped with a double $$, ie: $$(VAR_NAME).
     Escaped references will never be expanded, regardless of whether the
     variable exists or not. Cannot be updated. More info:
     https://kubernetes.io/docs/tasks/inject-data-application/define-command-argument-container/#running-a-command-in-a-shell```

Reply:
Maybe we should set the container to have an entry point?
Reason is that, the image can be replaced by the user even if he uses entrypoint.

Not really a hard requirement, just trying to shoot a lil more flexible approach here.

Reply:
Ok, got the idea. I'll do that.

Reply:
https://github.com/xridge/sysbench-docker/pull/2